src/sunmd5_fmt_plug.c

```diff
@@ -159,20 +159,11 @@ static struct fmt_tests tests[] = {
 /* output buffer for para is only 16 bytes per COEF, vs 64, so it's fewer bytes to jumbo to the next PARA start */
 #define PARAGETOUTPOS(i, index)		( (((index)&(SIMD_COEF_32-1))<<2) + (((i)&(0xffffffff-3))*SIMD_COEF_32) + ((i)&3) + (((unsigned int)index/SIMD_COEF_32*SIMD_COEF_32)<<4) )
 
-#if defined (_DEBUG)
-// for VC debugging
 JTR_ALIGN(MEM_ALIGN_SIMD) static unsigned char input_buf[BLK_CNT*MD5_CBLOCK];
 JTR_ALIGN(MEM_ALIGN_SIMD) static unsigned char out_buf[BLK_CNT*MD5_DIGEST_LENGTH];
 JTR_ALIGN(MEM_ALIGN_SIMD) static unsigned char input_buf_big[25][BLK_CNT*MD5_CBLOCK];
-/*  Now these are allocated in init()
-unsigned char input_buf[BLK_CNT*MD5_CBLOCK]         __attribute__ ((aligned(16)));
-unsigned char input_buf_big[25][BLK_CNT*MD5_CBLOCK] __attribute__ ((aligned(16)));
-unsigned char out_buf[BLK_CNT*MD5_DIGEST_LENGTH]    __attribute__ ((aligned(16)));
-*/
-#else
-static unsigned char *input_buf;
-static unsigned char *out_buf;
-static unsigned char (*input_buf_big)[BLK_CNT*MD5_CBLOCK];
+#ifdef _OPENMP
+#pragma omp threadprivate(out_buf, input_buf, input_buf_big)
 #endif
 
```

Comment:
I opt to use static allocation for those arrays, to make OpenMP code clearer. This might need discussion.